### PR TITLE
[Bugfix] raftstore: fix the bug when the channel is full in AutoSplitController.

### LINKED ISSUE #16716: TiKV store down when running workload
### What version of TiKV are you using?
1c7b34419feaf8901046df335d4da55ad3ca9b5d

### What operating system and CPU are you using?
X86 K8S

### Steps to reproduce
1. TiDB Cluster with 30 TiKV node, Titan on, cluster size ~70TB, run 3 workloads:   2 Big tables (15TB each table, row width ~9KB); 4000 small tables (row width ~1.7KB); 1 larger row workload  (row width: ~1MB)

### What did you expect?
TiKV store shoult not down

### What did happened?
After the cluster recently upgraded from 397d8d4894d06913c6bc72304fe1f83ba6fedd33 to 1c7b34419feaf8901046df335d4da55ad3ca9b5d, TiKV down issue is seen when run workload, and restored after workload is stopped.
img_v3_029d_968e0445-f6d4-47fb-9cd2-5c2cc841d34g

### PR DESCRIPTION
**Pull request by LykxSassinator**
### What is changed and how it works?

Issue Number: Close #16716

What's Changed:

```commit-message
Fix the bug when the channel is full in `AutoSplitController`.
```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by glorv — APPROVED** [review]

File: components/raftstore/src/store/worker/pd.rs [hunk 1]
```diff
@@ -778,17 +778,17 @@ where
     #[inline]
     pub fn maybe_send_read_stats(&self, read_stats: ReadStats) {
         if let Some(sender) = &self.read_stats_sender {
-            if sender.send(read_stats).is_err() {
-                debug!("send read_stats failed, are we shutting down?")
+            if sender.try_send(read_stats).is_err() {
```
Comment (glorv) [inline components/raftstore/src/store/worker/pd.rs:781]: Maybe also write the error into the log

Comment (LykxSassinator) [reply components/raftstore/src/store/worker/pd.rs:781]: inapposite. 
If the channel is full, and the logging will be unstoppable.

File: components/raftstore/src/store/worker/pd.rs [hunk 2]
```diff
@@ -778,17 +778,17 @@ where
     #[inline]
     pub fn maybe_send_read_stats(&self, read_stats: ReadStats) {
```
Comment (overvenus) [inline components/raftstore/src/store/worker/pd.rs:779]: Please add a test to make sure `maybe_send_read_stats` and `maybe_send_cpu_stats` is unblocking.

(You can set up a mock `StatsMonitor` in tests.)

**Review by overvenus — COMMENTED** [review]
Rest LGTM

**Review by tonyxuqqi — APPROVED** [review]

File: components/raftstore/src/store/worker/pd.rs (hunk 2 above)
Comment (LykxSassinator) [reply components/raftstore/src/store/worker/pd.rs:779]: Got it, I'll supplement an extra test case for it.

Comment (LykxSassinator) [reply components/raftstore/src/store/worker/pd.rs:779]: Ref: https://github.com/tikv/tikv/pull/16729

**Comment by overvenus** [comment]
Touch #16678

**Comment by overvenus** [comment]
Touch #17043

**Pull request merged**